added sem ver clarification - #850 (#855)
radiantearth/stac-spec: README.md

at this point we don't anticipate any major changes, but reserve the right to make them if we get feedback that something just doesn't 
work. Which is to say the next couple months are a great time to implement STAC, as your changes will be head. After 1.0 our goal
is to not change the core in any backwards incompatible way for a very long time, if ever, so that people can build on this until 

> JSON is no longer relevant. STAC spec aims to follow [Semantic Versioning](https://semver.org/), so once 1.0.0 is reached any breaking

Reviewer: ```suggestion
JSON is no longer relevant. The STAC specification follows [Semantic Versioning](https://semver.org/), so once 1.0.0 is reached any breaking
```

Author: The reason I said 'aims' is that I feel like pre-1.0 we haven't perfectly followed it. Though I guess this commit is just 'going forward', and our 1.0.0-beta.1 does follow it.

Reviewer: I think SemVer doesn't make any assumptions on breaking changes for 0.x versions (you can do whetever you want), so we are totally fine with the proposal.

Reviewer: Yep, proof is here: https://semver.org/#spec-item-4

Author: Cool.
